### PR TITLE
[XPU] rotary when use_neox_rotary_style = True, support single input and gqa

### PR DESCRIPTION
**Pull request by HarperCy**
### PR Category
Operator Mechanism

### PR Types
New features

### Description
rotary when use_neox_rotary_style = True, support single input and gqa

### CODE REVIEW

**Review by lj970926 — APPROVED** [review]
LGTM，后续可以考虑给GQA相关规模加个单测

**Comment by HarperCy** [comment]
> LGTM，后续可以考虑给GQA相关规模加个单测

收到

File: paddle/phi/kernels/fusion/xpu/fused_rope_grad_kernel.cc [hunk 1]
```diff
@@ -64,31 +64,61 @@ void FusedRopeGradKernel(const Context& dev_ctx,
     XPUGetSinCosData<XPUType, Context>(
         dev_ctx, cos, position_ids, cos_data, batch_size, seq_len, head_dim);
     if (!dout_k.get_ptr()) {
-      PADDLE_THROW(phi::errors::Unimplemented(
-          "XPU do not support rotary_embedding_grad without qk when "
-          "use_neox_rotary_style == True."));
+      auto* dq_data = reinterpret_cast<XPUType*>(dev_ctx.template Alloc<T>(dq));
```
Comment (QingshuChen) [inline paddle/phi/kernels/fusion/xpu/fused_rope_grad_kernel.cc:67]: 为什么之前EB3.5训练能正常执行, 这边没有报错?

**Review by QingshuChen — APPROVED** [review]

File: paddle/phi/kernels/fusion/xpu/fused_rope_grad_kernel.cc (hunk 1 above)
Comment (HarperCy) [reply paddle/phi/kernels/fusion/xpu/fused_rope_grad_kernel.cc:67]: Throw信息没考虑全，漏掉一个情况，没有触发报错

**Pull request merged**